bw2io/backup.py

```diff
@@ -55,40 +62,155 @@ def backup_data_directory(
     timestamp_str = (
         f'.{datetime.datetime.now().strftime("%d-%B-%Y-%I-%M%p")}' if timestamp else ""
     )
-    backup_filename = f"brightway2-data-backup{timestamp_str}.tar.gz"
+    backup_filename = f"brightway2-data-backup{file_suffix}{timestamp_str}.tar.gz"
     fp = dir_backup / backup_filename
 
     # Create the backup archive
-    print("Creating backup archive of data directory - this could take a few minutes...")
+    print(
+        "Creating backup archive of data directory - this could take a few minutes..."
+    )
     with tarfile.open(fp, "w:gz") as tar:
         data_directory = Path(projects._base_data_dir)
         tar.add(data_directory, arcname=data_directory.name)
 
     print(f"Saved to: {fp}")
-    
+
     return fp
 
 
+def restore_data_directory(
+    fp: Union[str, Path],
+    new_data_dir: Optional[Union[str, Path]] = None,
+):
+    """
+    Restores a brightway data directory from a tar.gz file. If the data directory already exists, you must confirm that you want to delete it.
+    If ``new_data_dir`` is specified, the data directory will be restored to that location. Otherwise, the data directory will be restored to the default location. e.g: ``~/.local/share/brightway3`` on Linux.
+
+    Parameters
+    ----------
+    fp (Union[str, Path]):
+        The file path of the tar.gz file.
+    new_data_dir (Optional[Union[str, Path]]):
+        The new data directory path.
+
+    Raises
+    ------
+    ValueError:
+        If the file doesn't exist
+    PermissionError:
+        If the data directory is not writable.
+    Exception:
+        If there's an attempted path traversal in the tar file.
+
+    Returns
+    -------
+    data_dir (Path):
+        The path of the restored data directory.
+
+    See Also
+    --------
+    bw2io.backup.backup_data_directory: To backup the data directory.
+    """
+
+    # check if file exists
+    fp = Path(fp)
```

Comment:
If we are doing this, then I don't see any reason to allow `str` inputs. I thought the point was to allow for non-filesystem resources, but I would prefer for all filepaths to be `Paths`. I have taken this approach in `bw2data`.